sos/plugins/cnv.py

```diff
@@ -0,0 +1,234 @@
+            options = self.get_option("vm")
+            if isinstance(options, bool):
+                return
+
+            subcmds_vm = [
+                "describe vm",
+                "describe vmi",
+                "get vm -o yaml",
+                "get vmi -o yaml",
+            ]
+            node_pods = [
+                "%s get pods -l kubevirt.io=virt-handler "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=multus "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=ovs-cni "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=ovs "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=sdn "
+                "--all-namespaces -o wide" % oc_cmd,
+            ]
+
+            for vm in self.get_vm_ns(vmis, self.get_option("vm")):
+                for s in subcmds_vm:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + vm])
+                getvmi = "%s get vmi " % oc_cmd + vm + " --show-labels"
+                lbl_ext = self.call_ext_prog(getvmi)
+                vmi_l = ""
+                if lbl_ext['status'] != 0:
+                    return
+                for line in lbl_ext['output'].splitlines():
+                    if line.split()[0] == vm.split()[0]:
+                        for l in line.split()[2].split(','):
+                            if vm.split()[0] in l:
+                                vmi_l = "-l " + l + " --all-namespaces -o wide"
+                p_ln = self.get_cmd_output_now("%s get pods " % oc_cmd + vmi_l)
+                for pod in self.get_resc_ns(p_ln):
+                    for s in subcmds_pods:
+                        self.add_cmd_output(["%s " % oc_cmd + s + " " + pod])
+                    exc = "%s " % oc_cmd + "exec " + pod + " -- virsh dumpxml "
+                    dmn = vm.split()[2] + "_" + vm.split()[0]
+                    self.add_cmd_output([exc + dmn])
+                node = self.get_node_info(p_ln)
+                vm_pods = []
+                for pod in node_pods:
+                    node_pod_ext = self.call_ext_prog(pod)
+                    if node_pod_ext['status'] == 0:
+                        for line in node_pod_ext['output'].splitlines():
+                            if line.startswith("NAMESPACE") \
+                                    or line.startswith("No resources found") \
+                                    or line.startswith("Error") \
+                                    or line.startswith("error"):
+                                pass
+                            else:
+                                if line.split()[7] == node:
+                                    p = line.split()[1]
+                                    ns = " -n " + line.split()[0]
+                                    vm_pods.append(p + ns)
+                for pod in vm_pods:
+                    for s in subcmds_pods:
```

Comment:
The kubernetes plugin already handles describe pods and pod logs via the `kubernetes.podlogs` option (disabled by default because it can get to be an absurd amount of output and we can't reliably obfuscate it all)